### PR TITLE
refactor(w3c/templates/sotd): contain its own l10n strings

### PR DESCRIPTION
**Pull request by ridhishjain**
refactor #2084
templates/sotd.js
tests are still to be added

### CODE REVIEW

**Comment by sidvishnoi** [comment]
@ridhishjain Please avoid sending too many PRs at once. If one of your PR is blocked for some reason, then work on other. Try to get an older one merged before sending other specially when they're related. This ensures your effort is not wasted and maintainers do not keep getting notifications.
Also, try using GitHub's "draft PR" feature, which lets us know it's not ready for review yet.

**Comment by sidvishnoi** [comment]
Thanks for working on this. We do need to remove the data from `core/l10n.js` as we move them to their respective files.

**Comment by ridhishjain** [comment]
Sorry for trouble, but can u tell me the corresponding test file for `src/w3c/templates/sotd.js`
The test files present in `tests/w3c` are corresponding to those of `src/w3c`. Where will I find test files related to the templates folder.......as there is no such folder in `tests/w3c`

**Comment by sidvishnoi** [comment]
@ridhishjain They're present in `tests/spec/w3c/headers-spec.js`

**Comment by ridhishjain** [comment]
@sidvishnoi review required

**Comment by marcoscaceres** [comment]
@ridhishjain, not a reprimand (I know you are new to open source), but please refrain from writing “review required”. We monitor all pull request so it’s not needed - just generates redundant emails.

File: tests/spec/w3c/headers-spec.js [hunk 1]
```diff
@@ -1582,4 +1583,25 @@ describe("W3C — Headers", () => {
...
   });
+
+  it("localizes sotd", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      htmlAttrs: {
+        lang: "es",
+      },
+      body: `
+        <section class="">
```
Comment (sidvishnoi) [inline tests/spec/w3c/headers-spec.js]: This section is not needed

File: src/w3c/templates/sotd.js [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+  nl: {
+    sotd: "Status van dit document",
+  },
+  es: {
+    sotd: "Estado de este Document",
+  },
+};
+
+const lang = defaultLang in localizationStrings ? defaultLang : "en";
+const l10n = localizationStrings[lang];
```
Comment (sidvishnoi) [inline src/w3c/templates/sotd.js]: Please incorporate the changes from https://github.com/w3c/respec/pull/2646, such as:
```suggestion
const l10n = getIntlData(localizationStrings);
```

**Comment by ridhishjain** [comment]
Is it
 OK, should I start working on the next [PR](https://github.com/w3c/respec/pull/2644) then?

**Comment by ridhishjain** [comment]
Removal is generating some kind of error ??
related to `respec2htrml`

**Comment by sidvishnoi** [comment]
> Removal is generating some kind of error ??

Only the `sotd` strings were supposed to be removed from l10n.js, as they're now moved to templates/sotd.js via this PR.

**Pull request merged**